ui/src/pages/MultiSampleH5adDatasetPage.vue

```diff
@@ -0,0 +1,159 @@
+<script setup lang="ts">
+import type {
+  ColDef,
+  GridOptions,
+} from 'ag-grid-enterprise';
+import {
+  ClientSideRowModelModule,
+  MenuModule,
+  ModuleRegistry,
+  RichSelectModule,
+} from 'ag-grid-enterprise';
+
+import { AgGridVue } from 'ag-grid-vue3';
+
+import type { DSMultiSampleH5ad, PlId } from '@platforma-open/milaboratories.samples-and-data.model';
+import type { ImportFileHandle } from '@platforma-sdk/model';
+import { AgGridTheme, makeRowNumberColDef, PlAgCellFile, PlAgColumnHeader } from '@platforma-sdk/ui-vue';
+import { computed } from 'vue';
+import { useApp } from '../app';
+import SyncDatasetDialog from '../dialogs/SyncDatasetDialog.vue';
+import { agGroupIdColumnDef } from '../util';
+
+ModuleRegistry.registerModules([ClientSideRowModelModule, RichSelectModule, MenuModule]);
+
+const app = useApp();
+
+const datasetId = app.queryParams.id;
+
+const dataset = computed(() => {
+  const ds = app.model.args.datasets.find((ds) => ds.id === datasetId);
+  if (!ds)
+    throw new Error('Dataset not found');
+  return ds as DSMultiSampleH5ad;
+});
+
+type DatasetRow = {
+  readonly groupId: PlId;
+  readonly groupLabel: string;
+  readonly nSamples: number | undefined;
+  readonly data?: ImportFileHandle | null;
```

Comment:
Also three-state field, if it is really used somewhere a short comment would be a useful breadcrumb for the reader.